gulpfile.js/tests/configGenerator.test.js

```diff
@@ -0,0 +1,123 @@
+var fs = require('fs')
+var test = require('tape')
+var configGenerator = require('./../util/backstop/configGenerator')
+var deepClone = require('./../util/deepClone')
+var testFixturesPath = './gulpfile.js/tests/fixtures/configGenerator/'
+var config = {
+  compLib: {
+    port: 7000,
+    host: 'http://example',
+    baseDir: testFixturesPath
+  },
+  vrt: {
+    backstopConfigTemplate: testFixturesPath + 'backstop.template.json',
+    backstopConfig: testFixturesPath + 'backstop.json'
+  }
+}
+
+var removeFiles = function (files) {
+  return new Promise(function (resolve) {
+    files.forEach(function (file) {
+      fs.unlink(file, resolve)
+    })
+  })
+}
+
+var getBackstopConfigData = function () {
+  return new Promise(function (resolve, reject) {
+    fs.readFile(config.vrt.backstopConfig, function (err, data) {
```

Comment:
Maybe.
`BuildScenarios` stream builds and writes the structure for the `backstop.json` file, there are separate `read` and `write` streams in [configGenerator](https://github.com/hmrc/assets-frontend/blob/master/gulpfile.js/util/backstop/configGenerator.js#L15-L33). `configGenerator` is the thing that combines all of these parts.